src/rgw/rgw_rados.cc

```diff
@@ -12987,6 +12988,10 @@ int RGWRados::cls_user_sync_bucket_stats(rgw_raw_obj& user_obj, const RGWBucketI
       entry.size += header_stats.total_size;
       entry.size_rounded += header_stats.total_size_rounded;
       entry.count += header_stats.num_entries;
+      r = check_bucket_shards(bucket_info, bucket_info.bucket, bucket_quota);
+      if (r < 0) {
+	ldout(cct, 0) << "ERROR in check_bucket_shards: " << cpp_strerror(-r)<< dendl;
+      }
```

Comment:
@oritwas shouldn't have this inside the loop here, need to move it outside. Actually, I think this is the wrong method to put this in, maybe `rgw_user.cc:rgw_user_sync_all_stats()` is better?